sdcv.el

```diff
@@ -285,7 +286,17 @@ Voice will fetch from youdao.com if you use other system."
 (defcustom sdcv-env-lang "zh_CN.UTF-8"
   "Default LANG environment for sdcv program.
 
-Default is zh_CN.UTF-8, maybe you need change to other coding if your system is not zh_CN.UTF-8."
+Default is zh_CN.UTF-8, maybe you need change to
+other coding if your system is not zh_CN.UTF-8."
+  :type 'string
+  :group 'sdcv)
+
+(defcustom sdcv-command-string  "env LANG=%s %s -x -n %s %s --utf8-output --utf8-input --data-dir=%s"
```

Comment:
I think a list would be preferable to a format string. Since it is less error-prone.

I imagine it would be a a list of strings and symbols. For example:
```
'(program "-x" "-n" dictionary-list word "--utf8-output" "--utf8-input" "--data-dir" dictionary-data-dir)
```
It is very easy to create a **customize** interface for such list.

At runtime, one would cycle through this list and swap special symbols with values. Instead of `shell-command-to-string` something like `process-file` would be used.

Environment variables will be added to process-environment temporarily. 

By doing this way, problems with shell quoting would be avoided.

Reply:
That's a good point, back then  was a elisp scrub looking for a quick dirty fix and this is what I came up with :rofl: .

Look at this PR again I realised that I might over looked the compatibility aspect, I shouldn't replace the
obsolete alias for example.

I think this PR need to be redone. If you don't mind, can you do another PR on this command-string topic ?
I'll open another one deal with the obsolete function name and cl-lib.

Reply:
That wood be a good idea. I'm just not sure if it is better to go with this solution, or maybe use something like `format-spec`.